app/frontend/controllers/app_controller.js

```diff
@@ -173,10 +176,10 @@ class AppController {
     }
   }
 
-  initVerseButtons(tabIndex=undefined) {
-    var currentVerseListMenu = app_controller.getCurrentVerseListMenu(tabIndex)[0];
-    var editNoteButton = currentVerseListMenu.querySelector('.edit-note-button');
-    var copyButton = currentVerseListMenu.querySelector('.copy-clipboard-button');
+  initVerseContextButtons() {
+    var verseContextMenu = document.getElementById('verse-context-menu');
+    var editNoteButton = verseContextMenu.querySelector('.edit-note-button');
+    var copyButton = verseContextMenu.querySelector('.copy-clipboard-button');
```

Comment:
How about moving verseContextMenu into a separate component along with `initVerseContextMenu()`, `toggleVerseContextMenuButton()`, `enableVerseButtons()`, `disableVerseButtons()` and `hideVerseContextMenu()` methods?

Reply:
Yes, that makes sense :+1:

Reply:
I moved the verse context menu into a separate component now. I kept it without shadow-dom, because that was the easiest. And the menu button is still separate. It's not perfect this way, but I think sufficient for now. Feel free to have another look.